vm/interpreter/src/vm.rs

```diff
@@ -262,11 +258,11 @@ where
 
     /// Applies the state transition for a single message
     /// Returns ApplyRet structure which contains the message receipt and some meta data.
-    pub fn apply_message(&mut self, msg: &UnsignedMessage) -> Result<ApplyRet, String> {
-        check_message(msg)?;
+    pub fn apply_message(&mut self, msg: &ChainMessage) -> Result<ApplyRet, String> {
+        check_message(msg.message())?;
 
         let pl = price_list_by_epoch(self.epoch());
-        let ser_msg = &msg.marshal_cbor().map_err(|e| e.to_string())?;
+        let ser_msg = forest_encoding::to_vec(msg).map_err(|e| e.to_string())?;
```

Comment:
I made this change, but checking back it should be `msg.marshal_cbor` as it was (there is a very slight difference in this case, because if ever a BLS signed message was passed in, it should only charge for the unsigned message length. This doesn't happen but should be this way